distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso

```diff
@@ -1173,11 +1174,19 @@ type Column
          Replace texts in quotes with parentheses.
 
              column.replace '"(.*?)"'.to_regex '($1)'
-    replace : Text | Column | Regex -> Text | Column -> Case_Sensitivity -> Boolean -> Column
+    @term make_regex_text_widget
+    replace : Text | Regex | Column -> Text | Column -> Case_Sensitivity -> Boolean -> Column ! Unsupported_Database_Operation
     replace self term="" new_text="" case_sensitivity=Case_Sensitivity.Default only_first=False =
         Value_Type.expect_text self <| case_sensitivity.disallow_non_default_locale <|
             input_type = Meta.type_of term
-            self.make_op "REPLACE" [term, new_text] "new name" [input_type, case_sensitivity, only_first]
+            params = Replace_Params.Value input_type case_sensitivity only_first
+            case self.connection.dialect.are_replace_params_supported params of
+                False -> params.throw_unsupported
+                True ->
+                    raw_term = case term of
+                        _ : Regex -> term.pattern_string
+                        _ -> term
+                    self.make_op "REPLACE" [raw_term, new_text] "new name" [term, params]
```

Comment:
Ah, I see - because you use the same type for checking 'is supported'.

Ok I think that is good, personally I think I'd still create a `Replace_Metadata` type to hold this pair, but a 2-element Vector is good enough.

Reply:
No, because we use Replace_Params to define the parameters that are supported. We have lists of supported values of Replace_Params which we use to determine whether or not a combination is supported, and what to expect when running tests. The term can't be a part of that.